### ISSUE TITLE
508-defect-2 [SCREENREADER]: Search should not be fired upon selecting a typeahead suggestion from the list with `enter`

### ISSUE THREAD

**Issue by AngelaFowler82** [department-of-veterans-affairs/va.gov-team#24779]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**VFS Point of Contact:** Angela

## User Story or Problem Statement:
As a screen reader user, I expect to be able to select a suggestion before firing off a search.

## Details
When I type in a search string and arrow through the suggestions, NVDA announces that each item is unselected. However pressing "enter" fires off a search instead of selecting the suggestion.

## Suggested solution

**Note from Josh:**
I believe we're currently using [WAI-ARIA's 1.2's combobox pattern](https://www.w3.org/TR/wai-aria-practices-1.2/examples/combobox/combobox-autocomplete-list.html) with downshift. In that case users will expect the combobox portion of the pattern to _only be used to select items_, not fire off a search as it will be announced as such through default screen reader instructions. 

There's also the tested concern of folks expecting that `tab` will allow them select items from the suggestions list instead of an explicit `enter` as [documented in Sarah Higley's usability select and combobox usability tests](https://www.24a11y.com/2019/select-your-poison-part-2/#select-poison-recommendations).

I believe the team decided to pursue a search on `enter` from the suggestions list because it would allow us to track both what was written into the search and what was selected from the suggestions list. **I recommend that we look into this more to see if we can find a solution that allows for tracking while still not changing the semantics of the select interaction for screen reader users.**

## Acceptance Criteria
- [ ] A search does not fire off selecting a suggestion from the typeahead list

## Environment
NVDA in particular makes this interaction more confusing as it will announce to the user that the suggestions are _unselected_ implying that using `enter` will only select them, not fire off a search. This confusing behavior was noted in:
- NVDA with Chrome
- NVDA with Firefox

**Comment by denisecoveyduc** [comment]
@joshkimux @AngelaFowler82 sharing this write up that describes our downshift implementation - let us know if you'd like to discuss at all. cc: @srsuddath 

- - - 

Our current implementation of Typeahead has a couple key requirements that resulted in our implementation choices.

we need to be able to track the suggestions a user receives, what their query is, and whether they used the suggestions presented
we need to meet accessibility standards for usability.
The original proposed implementation for Typeahead was to use a Datalist, however this component does not allow for custom event handlers, which did not meet many of our analytics requirements for this feature.

As a result, we had two options:

The first was to create a custom component to be owned and maintained by va.gov to implement typeahead in such a way that would meet analytics and accessibility requirements
The second was to use the 3rd party component DownShift, which had previously been used on VA.gov for a typeahead experience.
Due to this product being a POC, the decision was made to use Downshift, rather than sped the time required to build a custom component from scratch, saving us a LOT of time in approvals and streamlining the release process.

However, Downshift lacks a large number of customization options, and obfuscates a lot of functionality from the developer (certain elements can't have event listeners added to them, or cannot be interacted with from outside of the component). It is because of these quirks that many of the "Nice to have" features are not an option for us currently.

Additionally, because the downshift component hides a lot of its interior elements, there are some functionalities that might "flake out" due to the component loading, such as focusing the input field on clicking the search menu button.

There are talks about creating a unified sitewide search, and as part of that process a more robust "custom" typeahead component would be designed and built that would allow for many of these features to exist.

**Comment by joshkimux** [comment]
Reopening this and attaching it to the full 508 audit epic for awareness. I'll remove this from the staging review 👍

**Comment by denisecoveyduc** [comment]
@joshkimux @AngelaFowler82  The team and I were discussing this ticket and wanted to circle back on this issue from a while ago.

We are currently iterating on the typeahead experience and invision an experience where pressing "enter" fires off a search  @srsuddath is currently working on a drop down component for typeahead which will replace the typeahead in the header.  He should have a review instance for that in the coming sprint and that drop down component will also go thru the full 508 Audit. 

As a result, I am wondering if we can close this issue to wrap up the "typeahead 1.0" initiative - your review of typeahead 2.0 (aka the dropdown component) you can see how the experience functions for screenreader users. Agree?

**Comment by joshkimux** [comment]
@denisecoveyduc Thank you for filling us in on the progress-- it's super exciting to hear a new typeahead will be coming around soon for review.

I agree that we can close this issue to wrap up the typeahead 1.0 initiative once typeahead 2.0 comes around. Just for our own tracking purposes though, would you be comfortable with us holding off closing this until the typeahead 2.0 a11y review is officially completed? 🙇

**Comment by AngelaFowler82** [comment]
This issue should not be closed until and unless it has been resolved or the issues outlined therein have been moved to another ticket.

**Comment by srsuddath** [comment]
We are closing this ticket as typeahead 2.0 has officially been launched
